src/lib/createGHA/index.ts

```diff
@@ -0,0 +1,234 @@
+
+const GITHUB_SECRET_NAME = 'README_API_KEY';
+
+export const git = simpleGit();
+
+/**
+ * Removes any non-file-friendly characters and adds
+ * the full path + file extension for GitHub Workflow files.
+ * @param fileName raw file name to clean up
+ */
+export const cleanUpFileName = (fileName: string) => {
+  return path.join('.github/workflows/', `${fileName.replace(/[^a-z0-9]/gi, '-').toLowerCase()}.yml`);
+};
+
+/**
+ * Constructs the command string that we pass into the workflow file.
+ */
+function constructCmdString(
+  command: keyof typeof commands,
+  args: OptionDefinition[],
+  opts: CommandOptions<Record<string, string | boolean | undefined>>
+): string {
+  const optsString = args
+    .sort(arg => (arg.defaultOption ? -1 : 0))
+    .map(arg => {
+      const val = opts[arg.name];
+      // if default option, return the value
+      if (arg.defaultOption) return val;
+      // obfuscate the key in a GitHub secret
+      if (arg.name === 'key') return `--key=$\{{ secrets.${GITHUB_SECRET_NAME} }}`;
+      // remove the GitHub flag
+      if (arg.name === 'github') return false;
+      // if a boolean value, return the flag
+      if (arg.type === Boolean && val) return `--${arg.name}`;
+      if (val) return `--${arg.name}=${val}`;
+      return false;
+    })
+    .filter(Boolean)
+    .join(' ');
+
+  return `${command} ${optsString}`.trim();
+}
+
+/**
+ * Function to return various git attributes needed for running GitHub Action
+ */
+export async function getGitData() {
+  // Expressions to search raw output of `git remote show origin`
+  const headRegEx = /^ {2}HEAD branch: /g;
+  const headLineRegEx = /^ {2}HEAD branch:.*/gm;
+
+  const isRepo = await git.checkIsRepo().catch(e => {
+    debug(`error running git repo check: ${e.message}`);
+    return false;
+  });
+
+  let containsGitHubRemote;
+  let containsNonGitHubRemote;
+  let defaultBranch;
+  const rawRemotes = await git.remote([]).catch(e => {
+    debug(`error grabbing git remotes: ${e.message}`);
+    return '';
+  });
+
+  if (rawRemotes) {
+    const remote = rawRemotes.split('\n')[0];
+    const rawRemote = await git.remote(['show', remote]);
+    // Extract head branch from git output
+    const rawHead = headLineRegEx.exec(rawRemote as string)?.[0];
+    if (rawHead) defaultBranch = rawHead.replace(headRegEx, '');
+
+    // Extract the word 'github' from git output
+    const remotesList = (await git.remote(['-v'])) as string;
+    // This is a bit hairy but we want to keep it fairly general here
+    // in case of GitHub Enterprise, etc.
+    containsGitHubRemote = /github/.test(remotesList);
+    containsNonGitHubRemote = /gitlab/.test(remotesList) || /bitbucket/.test(remotesList);
+  }
+
+  const repoRoot = await git.revparse(['--show-toplevel']).catch(e => {
+    debug(`error grabbing git root: ${e.message}`);
+    return '';
+  });
+
+  return { containsGitHubRemote, containsNonGitHubRemote, defaultBranch, isRepo, repoRoot };
+}
```

Comment:
@RyanGWU82 here's where we're running those git commands! The idea is that we want to automatically determine if we should ask the user if they'd like to generate a new GitHub Action workflow. See here for the business logic!

In order to do this, we'll need a few bits of information:

| Data We Need | Corresponding `git` command |
| :-------------: | :-------------:
| Flag to indicate if the repo is connected to GitHub | `git remote -v` |
| Flag to indicate if the repo is connected to **a non-GitHub remote** | `git remote -v` |
| What the default branch for the repo is | `git remote` and then `git remote show origin`[^1] |
| What the root for the repo is in the current directory | `git rev-parse --show-toplevel` |

Any and all feedback on this approach would be appreciated 😇 

[^1]: `origin` in the second command is pulled dynamically from the first line of the output of the first command, and then the `HEAD` branch is grabbed via some regex 😬

Reply:
This all sounds great — I didn't even think it was possible to find the default branch on GitHub from a checkout.

"Whether or not this is a git-enabled repo" can be answered with `git rev-parse --show-toplevel` — if that succeeds then you're in a git-enabled repo. (Although relying on simple-git sounds fine too!)

(Haven't looked through the code yet) Are you using simple-git for all of this? Looks like it shells out to `git` on the user's system and it doesn't require a minimum git version? It might be a good idea to check to see if they have at least a somewhat modern version, just to avoid things going too badly…

Reply:
You might also want to just confirm that the `git` command is available locally and give a friendly error if it's not, or catch any/all `simple-git` errors and replace them with friendlier messages.

Reply:
> "Whether or not this is a git-enabled repo" can be answered with git rev-parse --show-toplevel — if that succeeds then you're in a git-enabled repo. (Although relying on simple-git sounds fine too!)

I believe [`isRepo()`](https://github.com/steveukx/git-js#git-rev-parse--repo-properties) is using this! But yeah I figured it was just easier to use `simple-git` since it's basically just Node syntactical sugar for `git` commands.

> You might also want to just confirm that the git command is available locally and give a friendly error if it's not, or catch any/all simple-git errors and replace them with friendlier messages.

For some add'l context, all of this data is purely optional and for the purpose of enhancing the onboarding experience (e.g. to determine if the working directory a GitHub repo or not), so I wrote [the code](https://github.com/readmeio/rdme/blob/06895f474a79edd68b38efbd28dd4a6b7f12ad74/src/lib/createGHA/index.ts#L80-L119) in such a way that everything falls back to falsy values that we just ignore. Say a user is not in a repo or has an outdated version of `git`, we just hit [this conditional](https://github.com/readmeio/rdme/blob/06895f474a79edd68b38efbd28dd4a6b7f12ad74/src/lib/createGHA/index.ts#L137) and [exit the function](https://github.com/readmeio/rdme/blob/06895f474a79edd68b38efbd28dd4a6b7f12ad74/src/lib/createGHA/index.ts#L147-L148) 🤷🏽‍♂️